test/version_solver_test.dart

```diff
@@ -1040,7 +1040,62 @@ void dartSdkConstraint() {
     await expectResolves(environment: {'_PUB_TEST_SDK_VERSION': '2.0.0'});
   });
 
-  test('package deps disallow 2.0.0-dev by default', () async {
+  test('package deps allow 2.0.0-dev by default', () async {
+    await d.dir('foo', [
+      await d.pubspec({'name': 'foo'})
+    ]).create();
+    await d.dir('bar', [
+      await d.pubspec({'name': 'bar'})
+    ]).create();
+
+    await d.dir(appPath, [
+      await d.pubspec({
+        'name': 'myapp',
+        'dependencies': {
+          'foo': {'path': '../foo'},
+          'bar': {'path': '../bar'},
+        }
+      })
+    ]).create();
+
+    await expectResolves(
+      environment: {'_PUB_TEST_SDK_VERSION': '2.0.0-dev.99'},
+      // Log output should mention the PUB_ALLOW_RELEASE_SDK environment
+      // variable and mention the foo and bar packages specifically.
+      output: allOf(contains('PUB_ALLOW_PRERELEASE_SDK'),
+          anyOf(contains('foo, bar'), contains('bar, foo'))),
+    );
+  });
+
+  test(
+      'pub doesn\'t log about pre-release sdk overrides if '
```

Comment:
Nit: I think it's a little cleaner to use a double-quoted string instead of escaping the apostrophe.

Reply:
done